infrahub_sdk/store.py

```diff
@@ -20,130 +23,336 @@ def get_schema_name(schema: str | type[SchemaType] | None = None) -> str | None:
     return None
 
 
-class NodeStoreBase:
-    """Internal Store for InfrahubNode objects.
-
-    Often while creating a lot of new objects,
-    we need to save them in order to reuse them later to associate them with another node for example.
-    """
+class NodeStoreBranch:
+    def __init__(self, name: str) -> None:
+        self.branch_name = name
 
-    def __init__(self) -> None:
-        self._store: dict[str, dict] = defaultdict(dict)
-        self._store_by_hfid: dict[str, Any] = defaultdict(dict)
+        self._objs: dict[str, InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync] = {}
+        self._hfids: dict[str, dict[tuple, str]] = defaultdict(dict)
+        self._keys: dict[str, str] = {}
+        self._uuids: dict[str, str] = {}
 
-    def _set(self, node: InfrahubNode | InfrahubNodeSync | SchemaType, key: str | None = None) -> None:
-        hfid = node.get_human_friendly_id_as_string(include_kind=True)
+    def count(self) -> int:
+        return len(self._objs)
 
-        if not key and not hfid:
-            raise ValueError("Cannot store node without human friendly ID or key.")
+    def set(self, node: InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync, key: str | None = None) -> None:
+        self._objs[node._internal_id] = node
 
         if key:
-            node_kind = node._schema.kind
-            self._store[node_kind][key] = node
+            self._keys[key] = node._internal_id
 
-        if hfid:
-            self._store_by_hfid[hfid] = node
+        if node.id:
+            self._uuids[node.id] = node._internal_id
+
+        if hfid := node.get_human_friendly_id():
+            self._hfids[node.get_kind()][tuple(hfid)] = node._internal_id
+
+    def get(  # type: ignore[no-untyped-def]
```

Comment:
Looks like you've fixed the problem with `# type: ignore[no-untyped-def]` so you can remove that part.